src/features/validation/nodeValidation/NodeValidation.tsx

```diff
@@ -0,0 +1,112 @@
+
+import { useAttachments } from 'src/features/attachments/AttachmentsContext';
+import { useCurrentLanguage } from 'src/features/language/LanguageProvider';
+import { Validation } from 'src/features/validation/validationContext';
+import { useAsRef } from 'src/hooks/useAsRef';
+import { useMemoDeepEqual } from 'src/hooks/useStateDeepEqual';
+import { implementsAnyValidation, implementsValidateComponent, implementsValidateEmptyField } from 'src/layout';
+import { useNodes } from 'src/utils/layout/NodesContext';
+import type { CompTypes } from 'src/layout/layout';
+import type { IComponentFormData } from 'src/utils/formComponentUtils';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
+
+export function NodeValidation() {
+  const nodes = useNodes().allNodes();
+  const nodesToValidate = nodes.filter(
+    (node) => implementsAnyValidation(node.def) && !('renderAsSummary' in node.item && node.item.renderAsSummary),
+  );
+
+  return (
+    <>
+      {nodesToValidate.map((node) => (
+        <SpecificNodeValidation
+          key={node.item.id}
+          node={node}
+        />
+      ))}
+    </>
+  );
+}
+
+function SpecificNodeValidation<Type extends CompTypes>({ node: _node }: { node: LayoutNode<Type> }) {
+  const updateComponentValidations = Validation.useUpdateComponentValidations();
+  const removeComponentValidations = Validation.useRemoveComponentValidations();
+  const nodeId = _node.item.id;
+  const validationDataSources = useValidationDataSourcesForNode(_node);
+
+  const nodeRef = useAsRef(_node);
+
+  useEffect(() => {
+    const node = nodeRef.current;
+
+    const validations: ComponentValidations[string] = {
+      component: [],
+      bindingKeys: node.item.dataModelBindings
+        ? Object.fromEntries(Object.keys(node.item.dataModelBindings).map((key) => [key, []]))
+        : {},
+    };
+
+    /**
+     * Run required validation
+     */
+    if (implementsValidateEmptyField(node.def)) {
+      for (const validation of node.def.runEmptyFieldValidation(node as any, validationDataSources as any)) {
+        if (validation.bindingKey) {
+          validations.bindingKeys[validation.bindingKey].push(validation);
+        } else {
+          validations.component.push(validation);
+        }
+      }
+    }
+
+    /**
+     * Run component validation
+     */
+    if (implementsValidateComponent(node.def)) {
+      for (const validation of node.def.runComponentValidation(node as any, validationDataSources as any)) {
+        if (validation.bindingKey) {
+          validations.bindingKeys[validation.bindingKey].push(validation);
+        } else {
+          validations.component.push(validation);
+        }
+      }
+    }
+
+    updateComponentValidations(nodeId, validations);
+  }, [nodeId, nodeRef, updateComponentValidations, validationDataSources]);
+
+  // Cleanup on unmount
+  useEffect(() => () => removeComponentValidations(nodeId), [nodeId, removeComponentValidations]);
```

Comment:
I see these now in multiple of these validation providers - but are they really needed? Won't the entire store be removed when navigating to a new task anyway - so there shouldn't be a need for setting a blank state in a context that is going away?

Reply:
You may be right, I think I just added this without testing that it was necessary

Reply:
Yup, that's what I was wondering. :grin: